### PR TITLE
Add check nan inf tools doc

### PR DESCRIPTION
**Pull request by wangxicoding**
We rewrite check nan inf tools in this PR https://github.com/PaddlePaddle/Paddle/pull/21076
Add check_nan_inf_cn document to this tools。

### CODE REVIEW

**Comment by wangxicoding** [comment]
中文文档，可以预览
https://github.com/PaddlePaddle/FluidDoc/blob/cd7dbdfd77cf3b03223ce9ac3c1a07f6e488b6ac/doc/fluid/flags/check_nan_inf_cn.md
或者查看截图
image

File: doc/fluid/flags/check_nan_inf_cn.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+nan - nan = nan, inf - inf = nan, num - num = 0,
+nan + nan = nan, inf + inf = inf, nan + 0 = nan,
+inf + 0 = inf, nan + inf = nan, 0 + 0 = 0
+```
+基于此使用如下操作仅需最后检查sum是否为nan或者inf就行了。
+```
+for(value:values): sum += (value-value)
+```
+
+***`注意`：本文档的进阶使用、速度、原理目前仅在develop的paddle版本生效。***
```
Comment (gongweibao) [inline doc/fluid/flags/check_nan_inf_cn.md]: 本文档的进阶使用、速度、原理目前仅在develop的paddle版本生效，并将随1.7发布。

Comment (wangxicoding) [reply doc/fluid/flags/check_nan_inf_cn.md]: Done.

**Review by gongweibao — APPROVED** [review]
LGTM

**Review by Boyan-Liu — APPROVED** [review]
LGTM

**Review by Boyan-Liu — CHANGES_REQUESTED** [review]
麻烦提交对应的英文文档。

**Comment by wangxicoding** [comment]
英文文档，可以预览
https://github.com/PaddlePaddle/FluidDoc/blob/cd7dbdfd77cf3b03223ce9ac3c1a07f6e488b6ac/doc/fluid/flags/check_nan_inf_en.md
或者截图
image

File: doc/fluid/flags/check_nan_inf_en.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+#### 2. CPU log information
+The nan, inf, and num printed in the CPU are displayed in front of the error message stack.
+The three values of nan, inf, and num are also printed, and the number of nan, inf, and num is printed.
+The name of the op and tensor which has nan or inf will be displayed in the end.
+![cpu_nan_inf.png](check_nan_inf_files/cpu_nan_inf.png)
+![cpu_nan_inf_op_var.png](check_nan_inf_files/cpu_nan_inf_op_var.png)
+
+## <span id="speed">Speed</span>
+Test environment: v100 32G single card, Resnet50 model, imagenet dataset.
+`The speed may be different under different environment and different model datasets. The following speeds are only for reference`
```
Comment (gongweibao) [inline doc/fluid/flags/check_nan_inf_en.md]: environments.

File: doc/fluid/flags/check_nan_inf_en.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# check nan inf tool
```
Comment (gongweibao) [inline doc/fluid/flags/check_nan_inf_en.md]: check_nan_inf tool?

File: doc/fluid/flags/check_nan_inf_en.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+nan - nan = nan, inf - inf = nan, num - num = 0,
+nan + nan = nan, inf + inf = inf, nan + 0 = nan,
+inf + 0 = inf, nan + inf = nan, 0 + 0 = 0
+```
+Based on this, using the following operation and only check the sum is nan or inf is enough.
+```
+for(value:values): sum += (value-value)
+```
+
+***`Note`: The Advanced use, Speed, and Principles of this document are currently only effective in the develop version of the Paddle, and will be released with the 1.7 version of the Paddle.
```
Comment (gongweibao) [inline doc/fluid/flags/check_nan_inf_en.md]: The Advanced use: Speed, and Principles

File: doc/fluid/flags/check_nan_inf_en.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+#### 2. CPU log information
+The nan, inf, and num printed in the CPU are displayed in front of the error message stack.
+The three values of nan, inf, and num are also printed, and the number of nan, inf, and num is printed.
+The name of the op and tensor which has nan or inf will be displayed in the end.
+![cpu_nan_inf.png](check_nan_inf_files/cpu_nan_inf.png)
+![cpu_nan_inf_op_var.png](check_nan_inf_files/cpu_nan_inf_op_var.png)
+
+## <span id="speed">Speed</span>
+Test environment: v100 32G single card, Resnet50 model, imagenet dataset.
```
Comment (gongweibao) [inline doc/fluid/flags/check_nan_inf_en.md]: Imagenet

File: doc/fluid/flags/check_nan_inf_en.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+`Note`: Op skip only accepts exact matches. To skip the softmax_with_cross_entropy check, you cannot set the environment variable to softmax_with or with_cross for fuzzy matching.
+You must set the full softmax_with_cross_entropy name.
+```
+export PADDLE_INF_NAN_SKIP_OP="mul"
+export PADDLE_INF_NAN_SKIP_OP="mul,softmax_with_cross_entropy"
+```
+##### 2.2 Skip op role check
+The currently accepted types are: forward, backward, optimize, rpc, dist, lrsched, loss, default.
+In fp32 training, it is not necessary to skip the nan inf check of the op role.
+However in `fp16` training, inf will be corrected in the back propagation, so it is generally necessary to skip the backward check, which is also the reason of add this function.
```
Comment (gongweibao) [inline doc/fluid/flags/check_nan_inf_en.md]: backpropagation

Comment (gongweibao) [inline doc/fluid/flags/check_nan_inf_en.md]: addition? of adding?

File: doc/fluid/flags/check_nan_inf_en.md (hunk 2 above)
Comment (wangxicoding) [reply doc/fluid/flags/check_nan_inf_en.md]: done.

File: doc/fluid/flags/check_nan_inf_en.md (hunk 4 above)
Comment (wangxicoding) [reply doc/fluid/flags/check_nan_inf_en.md]: Speed, and Principles do not belong to Advanced use.

File: doc/fluid/flags/check_nan_inf_en.md (hunk 5 above)
Comment (wangxicoding) [reply doc/fluid/flags/check_nan_inf_en.md]: done.

File: doc/fluid/flags/check_nan_inf_en.md (hunk 6 above)
Comment (wangxicoding) [reply doc/fluid/flags/check_nan_inf_en.md]: done

Comment (wangxicoding) [reply doc/fluid/flags/check_nan_inf_en.md]: done

Comment (wangxicoding) [reply doc/fluid/flags/check_nan_inf_en.md]: Chang to: which is why this feature is added.

**Review by gongweibao — APPROVED** [review]
LGTM

**Comment by wangxicoding** [comment]
> 麻烦提交对应的英文文档。

Done.

**Review by Boyan-Liu — APPROVED** [review]

**Pull request merged**